### PR TITLE
Update the OpenTelemetry SDK version to 1.37.0

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts [hunk 1]
```diff
@@ -4,7 +4,7 @@ plugins {
 
 dependencies {
   implementation(project(":instrumentation-api"))
-  implementation("io.opentelemetry:opentelemetry-extension-incubator")
+  implementation("io.opentelemetry:opentelemetry-api-incubator")
```
Comment (laurit) [inline instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts:7]: is this dependency needed at all?

File: instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/trace/ApplicationSpan.java [hunk 2]
```diff
@@ -100,6 +100,20 @@ public Span addEvent(
...
+    return this;
+  }
+
+  @Override
+  @CanIgnoreReturnValue
+  public Span addLink(SpanContext spanContext, Attributes attributes) {
+    agentSpan.addLink(Bridging.toAgent(spanContext), Bridging.toAgent(attributes));
+    return this;
+  }
+
```
Comment (laurit) [inline instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/trace/ApplicationSpan.java:116]: this deviates a bit from our usual approach of adding bridging for new methods in a subclass, but here it is much easier to do it the way it is so I think it is fine.

**Review by laurit — APPROVED** [review]

File: instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts (hunk 1 above)
Comment (zeitlinger) [reply instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts:7]: yes, for `EventLogger`

Comment (laurit) [reply instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts:7]: this module does not use `EventLogger` as far as I can tell

Comment (zeitlinger) [reply instrumentation/runtime-telemetry/runtime-telemetry-java8/library/build.gradle.kts]: right, mixed that up with another module - removed

File: instrumentation/opentelemetry-api/opentelemetry-api-1.31/javaagent/build.gradle.kts [hunk 3]
```diff
@@ -4,22 +4,21 @@ plugins {
...
   implementation(project(":instrumentation:opentelemetry-api:opentelemetry-api-1.27:javaagent"))
 
-  testImplementation("io.opentelemetry:opentelemetry-extension-incubator")
+  testImplementation("io.opentelemetry:opentelemetry-extension-incubator:1.31.0-alpha")
 }
 
 configurations.configureEach {
   if (name == "testRuntimeClasspath" || name == "testCompileClasspath") {
     resolutionStrategy {
       force("io.opentelemetry:opentelemetry-api:1.31.0")
```
Comment (trask) [inline instrumentation/opentelemetry-api/opentelemetry-api-1.31/javaagent/build.gradle.kts:21]: @laurit should we add something similar in all of the opentelemetry-api-* javaagent modules?

File: instrumentation/runtime-telemetry/runtime-telemetry-java8/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java8/JarAnalyzer.java [hunk 4]
```diff
@@ -43,8 +43,7 @@ final class JarAnalyzer implements ClassFileTransformer {
 
   private static final Logger logger = Logger.getLogger(JarAnalyzer.class.getName());
 
-  private static final String EVENT_DOMAIN_PACKAGE = "package";
-  private static final String EVENT_NAME_INFO = "info";
+  private static final String EVENT_NAME_INFO = "package.info";
```
Comment (trask) [inline instrumentation/runtime-telemetry/runtime-telemetry-java8/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java8/JarAnalyzer.java:46]: 👍

File: instrumentation/opentelemetry-api/opentelemetry-api-1.37/javaagent/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+testing {
+  suites {
+    val incubatorTest by registering(JvmTestSuite::class) {
+      dependencies {
+        implementation("io.opentelemetry:opentelemetry-api-incubator:1.37.0-alpha")
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply instrumentation/opentelemetry-api/opentelemetry-api-1.37/javaagent/build.gradle.kts:26]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11084

**Pull request merged**